### ISSUE TITLE
[Feature request] Load map preview

### ISSUE THREAD

**Issue by BannZay** [ddnet/ddnet#2892]
Its hard to clean up the map folder (I have a lot of "testmap", "partIdea#136", etc) or find a map without their name.
Loading each map taking a lot of time...
Can we show a map preview - picture of the map ? 
Most likely will need setting to turn this feature off.

**Comment by heinrich5991** [comment]
You could use the `render_map` tool from libtw2.

https://github.com/heinrich5991/libtw2/tree/3a62d5c212bbd6ba24bc5b6d80da356538ed456e/render_map

**Comment by Robyt3** [comment]
Would be cool to have a map preview in the client.

This was previously discussed at teeworlds/teeworlds#1578. This issue was mostly about a preview in the server browser. But a preview in the editor browser would be useful for the issue described by @BannZay (together with #6272).

How to embed the map preview?

1. Embed image:
   - A) let map maker embed a preview image taken as a screenshot in the editor
   - B) let map maker embed arbitrary image from computer (would be technically possible in case A as well, by changing the code)


2. Render part of the map dynamically:
   - A) let map maker specify screenshot position and parameters
      - use game tile to specify location and parameters
      - use separate map item in the map file to specify location
   - B) always render specific location with default settings (this would also be the fallback for case A when no preview is specified by the map maker)
      - entire map zoomed out
      - spawn/start/flag location

**Comment by Robyt3** [comment]
CC @AssassinTee. Previewing maps in the editor file browser and server browser could eventually be possible with the map rendering refactoring, but the down-side of dynamically rendering the map is that we'd have to load the entire map and its resources to preview it.

**Comment by AssassinTee** [comment]
> CC @AssassinTee. Previewing maps in the editor file browser and server browser could eventually be possible with the map rendering refactoring, but the down-side of dynamically rendering the map is that we'd have to load the entire map and its resources to preview it.

We could totally do it once and take a Screenshot buffering the preview

**Comment by AssassinTee** [comment]
One question, should entities overlays be visible in the preview? I have some testmaps with basically **no visuals** which would benefit from it. On the other side I'd still see value in having them disabled and just show a map in it's full glorious design

**Comment by AssassinTee** [comment]
@Robyt3 Since the maprender tool got merged, I'd now propose a cache directory for images where we generate small thumbnails for the filebrowser and big ones for the serverbrowser in a background job on demand. Do you have a better idea?

**Comment by def-** [comment]
This sounds fine to me. I think just generate big ones though, since I guess the image generation is expensive, and scaling it down is relatively cheaper?

**Comment by heinrich5991** [comment]
@Jupeyy said we already have a map renderer in DDNet, so we could render the map using our existing renderer, but onto a surface. This seemed reasonable to me, and it would probably disentangle our map code if we'd do that.

**Comment by AssassinTee** [comment]
> [@Jupeyy](https://github.com/Jupeyy) said we already have a map renderer in DDNet, so we could render the map using our existing renderer, but onto a surface. This seemed reasonable to me, and it would probably disentangle our map code if we'd do that.

yes, this is what #12822 is for, but this would also need a followup. I am working into this direction

CC #12848
